### PR TITLE
Basic No Internet Check

### PR DESCRIPTION
**Pull request by say25**
Application no longer will crash without internet connection. Future
work could include storing JSON data locally in Global.swift on data
retrieval and retrieving stored data if any exists if unable to
retrieve data.

### CODE REVIEW

File: big-red-shuttle/Controllers/ScheduleViewController.swift [hunk 1]
```diff
@@ -31,7 +31,9 @@ class ScheduleViewController: UIViewController, UITableViewDelegate, UITableView
         
         // Schedule Bar
         scheduleBar = ScheduleBar(frame: CGRect(x: 0, y: 0, width: view.frame.width, height: barHeight))
-        scheduleBar.setUp(buttonsData: loopStop!.allArrivalsToday(), selected: 0)
+        if (loopStop != nil) {
+            scheduleBar.setUp(buttonsData: loopStop!.allArrivalsToday(), selected: 0)
+        }
```
Comment (anniexcheng) [inline big-red-shuttle/Controllers/ScheduleViewController.swift:36]: `if let _  = loopStop`

File: big-red-shuttle/Controllers/ScheduleViewController.swift [hunk 2]
```diff
@@ -59,14 +61,16 @@ class ScheduleViewController: UIViewController, UITableViewDelegate, UITableView
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
         
-        let nextArrivalsToday = loopStop!.nextArrivalsToday()
+        if (loopStop != nil) {
```
Comment (anniexcheng) [inline big-red-shuttle/Controllers/ScheduleViewController.swift:64]: same as line 34

File: big-red-shuttle/Controllers/ScheduleViewController.swift [hunk 3]
```diff
@@ -94,6 +98,9 @@ class ScheduleViewController: UIViewController, UITableViewDelegate, UITableView
     // MARK: - TableView Datasource Methods
     
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if (stops.count == 0) {
+            return 0
+        }
         return stops.count * loopStop!.allArrivalsToday().count
```
Comment (anniexcheng) [inline big-red-shuttle/Controllers/ScheduleViewController.swift:104]: `return (stops.count > 0) ? stops.count * loopStop!.allArrivalsToday().count : 0`

**Comment by anniexcheng** [comment]
yea in the future, we probably want to have some local storage and update system so we don't have to check for nil

**Comment by say25** [comment]
Well you could still have a situation where you download the app lose your internet connection and then immediately try to open it. You would have no scheduled information stored and thus still need to check for nil. Super edge case but the check nil is so low cost its not a big deal. will make above changes.

**Comment by anniexcheng** [comment]
Yea I meant we could have a backup json file in our app that is a last resort and show a message that the info displayed may be outdated

**Comment by say25** [comment]
hhhmm maybe that makes more sense to implement than this pull request?

**Comment by anniexcheng** [comment]
yea I think that's the better way to go.. feel free to do it on this pr if you're up to it :)

**Comment by say25** [comment]
I think you'd be better off doing it 😛

**Comment by anniexcheng** [comment]
haha ok!

**Pull request closed without merging**